### DISCUSSION TITLE
Network v15 OhSnap

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#7935] [General Q&A]
If you have any questions wrt the[ Network v15 OhSnap Upgrade and Lotus Releases Timeline](https://github.com/filecoin-project/lotus/discussions/7898), leave a comment here!

**Comment by Eliovp** [comment]
@jennijuju , I have a question.

Wouldn't it be much fairer if this only applied to new CC sectors? 
Sectors generated after the upgrade.

Because right now, this is a big finger to the people that were doing real deal sealing from the start.

So again...

**Wouldn't it be much fairer if this only applied to new CC sectors? 
Sectors generated after the upgrade.**

**Reply by RobQuistNL** [reply]
This very much.

Edit it;

- Allow only regular deals in CC sectors sealed before the network upgrade epoch.
- Allow all deals (so also verified) in CC sectors sealed after the network upgrade epoch.

This way its fair to everyone, and PL's wish of turning CC sectors into useable data will be provided for.

CC sectors have been garbage from day 1 and this was calculated in by everyone.

**Reply by jennijuju** [reply]
> CC sectors have been garbage from day 1 and this was calculated in by everyone.
👎 

I believe it's been made clear that CC sectors are also valuable to the network since the very beginning. When the storage market demand is low, pledging and maintaining committed capacities sectors is a way for a storage miner to provably demonstrate that they are reserving space for the network, and ready to put that storage into use by storing real data when there are available deal proposals appealing to SP. With this upgrade, SPs who have been maintaining CC sectors over the past year will be able to provide a more efficient and faster deal onboarding service to the clients.

**Reply by momack2** [reply]
Note that any v1 proofs sectors (those initially sealed in the first 4 months of the Network before v1.1 sectors were introduced) are not eligible for extension and therefore effectively _ineligible_ for SnapDeals upgrades (since eligible sectors need to have a lifetime long enough to fit the entire deal lifetime). So there is no special benefit to SpaceRace participants from SnapDeals (nor was participating in SpaceRace "cheating" - early participants helped bust through a ton of roadblocks!)

Putting the CC Sectors that have already been committed to the Filecoin network to work storing useful data is 100% inline with the goals and mission of the network, and seamless upgrade from CCSector to deals sector has been a goal since the inception.

**Reply by Eliovp** [reply]
Noted and thx for the clarification.
Finally a decent answer.

Still, I think CC sectors from before the upgrade should only be eligible for regular deals and not verified ones.
Of course, if you're trying to create the "Illusion" that the network is growing massively, this to the outside world.
Then, of course, this is the ideal way to manipulate everyone into thinking that the network is growing massively.

That said, as far as I know, verified deals were introduced later and were never mentioned in the original white papers.
So should also not be part of the original idea to be able to upgrade CC sectors into real deal sectors.

**Reply by RobQuistNL** [reply]
>> I believe it's been made clear that CC sectors are also valuable to the network since the very beginning. 
👎

Not really, the CC was there to show the world how much data _could_ be onboarded while there was still work being done on the deal market. Upgrading CC sectors was part of the plan - but not without the need to use any hardware. (Or just C2 in this case)

What this SnapDeals setup means, is that people that rented / stole (pyramid schemes) a bunch of hardware to seal 30 PiB of CC sectors and now only have 2 machines with storage running, will be rewarded by being able to 10x their storage to 300PiB with the fake DC applications that are pouring in (i know there's "strict" regulations but there's still people being manipulated there), and all that without needing any more sealing hardware.

It doesn't feel fair. If we can append a simple rule "no verified data in CC before the upgrade epoch", my concerns will be gone. Everyone (yes, including me) would be happy with seeing more actual data being onboarded that way (I really don't mind that 30PiB of CC turning into 30PiB of actual data, in fact I'd like it!). I just wouldn't like the big CC users on the network gain a 10x power burst with private / unretrievable data. Because thats what's going to happen if this rule is not applied.

**Reply by momack2** [reply]
[Verified data](https://spec.filecoin.io/algorithms/verified_clients/) (later rebranded "Filecoin Plus") have been part of the Filecoin Spec since well before network launch. Saying some ccsectors are eligible to store verified data and some aren't is not ...logical - it's up to the client what SPs they want to store with, and the requirements that they put on their FIL+ data storage. This isn't something the network can or should try to control. If some SPs don't want to or aren't set up to store deals (ex, they are in regions that are unappealing to FIL+ clients, or haven't invested in good deal-making setups that also offer good retrievability, or don't have a reliable deal-making reputation built up), they won't get any new benefits from SnapDeals. 🤷‍♀️

**Reply by Eliovp** [reply]
@RobQuistNL You're talking to a wall mate.

They do not see it, it's all woke, unicorns and rainbows. 

It's absolutely unbelievable.

Things like "they are in regions that are unappealing to FIL+" or "they haven't invested in good deal-making setups"
Are you kidding me? 🤦 🤦‍♂️ 


I honest to god hope it's going to happen, that some of those bigger "we only seal CC sectors to get as much block rewards as possible and we do not give a shit about the network or the project" now will onboard a shitload of verified deals "At a crazy rapid pace because no effort has to be done". 

I'm going to have a laugh first and then will tag you all and will keep it simple "Told you so....."
Exactly like the SpaceRace thing and the Slingshot thing "Told you so."

**Reply by RobQuistNL** [reply]
> Saying some ccsectors are eligible to store verified data and some aren't is not ...logical 

To me it is, because it gives people without the proper sealing hardware an advantage. And there's a lot of them.

>they won't get any new benefits from SnapDeals. 🤷‍♀️

Well thats the problem, they will! Just like the data that was lost in slingshot restore, there will be a bunch of "companies" applying for datacap, which will be allowed because its all too vague, and then they will store their own, unretrievable files, with their own miners, and turn their CC sectors into 10x more power, without needing to invest in any new hardware / network setup.

100% YES from me for snapping CC into storage.
100% NO from for for allowing verified data to go into those old sectors.

This way;
- There's no cheap, easy 10x network power increase
- There's CC that can be cheaply turned into actual storage
- We can still "preseal" CC to reserve for both verified and unverified data to increase the deal onboarding speed (which I think is a huge plus in this SnapDeal setup)

I really don't see why the network can't append that rule. The network also controls the amount of datacap used, and there's even people (dangerous in a decentralized world) deciding what is verified and what is not.

We append that rule, everyone's happy and the game can't be cheated. Unfortunate for early CC sector sealers (like myself!) who can't use them to upgrade to Fil+, but its the fairest option in my opinion.

**Reply by jennijuju** [reply]
@RobQuistNL what Molly was trying to clarify is that early CC sectors, have V1 proof and a max 540 days life time. so let's say, if you have a v1 proof CC sector pledged in Oct 2020, they will be expired in April, which is 1 month after the OhSnap upgrade. Now then minimal deal lifetime is 180 days, > 1 month, so none of the deal will be able to fit in these sectors. That's why its suggested that Snapdeals will mostly be in CC sectors that's being added after main net liftoff.

**Reply by jennijuju** [reply]
>  everyone's happy 

I doubt folks who started CC pledging b/c Fil+, which is[ introduced in Oct 2020](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0003.md), will be happy.

**Reply by stuberman** [reply]
I have already seen around 600 sectors (mostly with deals) expire. The question of whether to extend the lifespan of current sectors is not so simple. 

@RobQuistNL I am not sure I understand why you think using snap deals with verified data is a problem. Can you explain?

**Reply by RobQuistNL** [reply]
@stuberman what I mean is that a miner like this one; https://filfox.info/en/address/f0688165 can increase their 130 PiB to 1300 PiB with a single GPU.

**Reply by stuberman** [reply]
A single GPU AND 1.3 Billion USD (63 million FIL) AND finding a ton of verified deals. Is that really a risk we should be concerned about?
What I see happening is a lot of new large storage providers are getting ready to enter the Filecoin network, start sealing CC sectors in anticipation of accepting verified deals. This seems more than legitimate to me.

**Reply by rjan90** [reply]
@RobQuistNL No, they won´t be able to upgrade that amount of CC-sectors with a a single GPU, at least not in a couple of years.

The current specs for doing SnapUpgrades from my testing are as follow: ~192GiB RAM, and a GPU with over 8.4GB VRAM. Depending on which GPU you have, doing a Snap-Upgrade of one CC-sector will still take from 15 minutes to 1.5 hours depending on which GPU´s you have.

And in the event that they where going to upgrade all those sectors to verified deals, they would need to pledge 10x FILs.

**Reply by RobQuistNL** [reply]
Okay, thanks for the info, that changes things. I was under the impression that its just 1 single layer of PC1 and a single C2 - which is doable with a single EPYC machine with 1 GPU. The collateral part is a good point that I hadn't taken into consideration, as is the part about V1 / expired sectors. I figured since we're able to extend CC sectors this would bubble through.

So my concern is now much less of a concern. Still a concern, but its not as bad as I thought :)

**Reply by RobQuistNL** [reply]
BTW, @stuberman , in order to turn 2.419.306 sectors into verified deals you'll "only" need 3.628.959‬ FIL. Its still a lot, but much less than the 63 million FIL you mentioned (how did you get this number btw?)

**Reply by stuberman** [reply]
@RobQuistNL  The calculation I used is based on (130 PiB x 10 collateral) where 1PiB costs [4928 FIL](https://fgas.io/#) per PiB = 6,406,400 FIL x $20/Fil = $128 million (I miscalculated the first time with an extra decimal point.

**Reply by jennijuju** [reply]
> . I was under the impression that its just 1 single layer of PC1 and a single C2 - which is doable with a single EPYC machine with 1 GPU

@RobQuistNL the whole new proof is new thus the trusted setup,  you can find more info on that spec here https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0019.md#specification

**Reply by RobQuistNL** [reply]
Sealing 1 32 GiB sector with verified data requires 1,5 FIL though. Looks like the one I linked is indeed a 64 GB setup and not 32 so 6m fil makes sense :) 

I And yeah I read the PDF too @jennijuju :)

**Reply by dkkapur** [reply]
One more thing I'd like to flag -> SPs that are just here to abuse Fil+ are unlikely to leverage it for SnapDeals for older sectors, since the max lifetime of that sector is less than the max lifetime of a net new sector, thereby having a lower expected rewards value over time. It's much more likely that they will opt for a new sector that can earn for longer for the same collateral/effort to abuse the system. We're working on getting a lot more transparency around on-chain client and SP behavior on the Fil+ side and using this to push back against clients likely to be abusing the system. @RobQuistNL @Eliovp if you're particularly interested in this -> would love to chat at an upcoming governance call about some of the efforts and get your suggestions/feedback.

**Reply by stuberman** [reply]
@RobQuistNL  let me try the sector math for verified deals:
130 PiB = 32GiB x 4,259,840 sectors. 
4,259,840 x 1.5 FIL each  = 6,390,760 FIL x $21.20 (today's price) = $135M

Frankly, I have always seen the collateral cost as the major hurdle in sealing, the hardware and OPEX pales in comparison, especially when storing verified deals.

**Comment by Eliovp** [comment]
Unbelievable.
The naivety...

Hey, @jennijuju I bet you like this new upgrade don't you?

99% of the top 100 storage providers(garbage CC sectors) are in Asia.
Hey, look at this program here --> https://slingshot.filecoin.io/recovery 
"An incentive to restore lost data (not even going into detail about how the data was lost.......)"
How many of those (from a specific part of the world) are helping out to restore the data? (You can see the countries btw, just look......)

I know I'm just wasting time here by writing this, even though I'm right, just as I was right during the space-race period, how I was also right about the Slingshot program that was being cheated as well.

Woke, Unicorns, Rainbows, the world has gone mad.

**Reply by jennijuju** [reply]
If SP doesn't use CC sectors to store data, then they won't benefit from this upgrade or getting the 10x power you were describing.

**Reply by Eliovp** [reply]
Simple solution, everyone happy.

- CC Sectors before the upgrade are not eligible for verified deals.

There, problem solved.

**Reply by momack2** [reply]
Your personal, demeaning tone here is **_absolutely unacceptable_**, @Eliovp. I don't care if you are frustrated or have differing opinions, no one should have to deal with your rudeness and harassment. Shame on you for personally attacking @jennijuju when she is just trying to answer your questions about a FIP designed and approved by the community, not her. If you can't be polite and offer opinions respectfully, **get out**. We don't tolerate this sort of rudeness & disrespect in the Filecoin community. 

Please review the [Filecoin Code of Conduct](https://github.com/filecoin-project/community/blob/master/CODE_OF_CONDUCT.md) and **consider this an official warning**.

**Reply by Eliovp** [reply]
@momack2 , oh damn, an official warning. 
Is speaking out loud, telling the truth wrong these days?

Now I'm the one that feels attacked and offended, no seriously, isn't this a decentralized open-source project?

**Reply by dkkapur** [reply]
As a sidenote, @Eliovp - participation in the Restore effort requires manual validation and selection of individuals/entities, so using this as an example of who is participating/how they are participating/what people are interested in doing on the network is a red herring.

**Reply by Eliovp** [reply]
@dkkapur Hey, don't worry about it.

Today, I officially stopped caring (again).
"Offical warning, thumbs down emoji's, telling me what I can or can't do" 
Seriously?

If you read in between the lines, of everything I say, it's always because of concerns towards the project. 

Anyway, I will be back again when things go sideways, just to say "Told you so" (again).

**Comment by cryptowhizzard** [comment]
Still, there is a legit worry.

A lot of CC sectors (Where ever they reside on the planet) who were created in the SR can be upgraded this way in a very short timeframe without a lot of hardware to be a verified sector. With a lot of LDN requests incoming to make verified deals this might be worth consideration how we intend to handle that because the incentive to cheat only get's larger this way.

@dkkapur

**Reply by jennijuju** [reply]
re SR sectors in question, please see https://github.com/filecoin-project/lotus/discussions/7935#discussioncomment-2111910

**Reply by Angelo-gh3990** [reply]
I agree there are a lot of cc sectors out there, but they have a short useable life span... To short to make it worth to be used for snap deals I believe, it would be to much of a hassle to manage that over the potential benefit it will give.

I think most new cc sectors with a normal lifespan will be used for snap deals making this a very good solution to seal a deal for a customer. I'm just looking at the benefit from a customers perspective as they will have there data sealed very quick when you have cc sectors at the ready.

Some concern is around people that would do self dealing amongst themselves or with a small group of people, but they have to come up with a 10x collateral at that point to. I don't know if they will really go do that.

**Reply by dkkapur** [reply]
Thanks @cryptowhizzard - agree that it is a risk, but its lessened by sector max lifetimes + in general, I think people have been a lot more opportunistic in the last couple of months with LDNs and hoping to make the most of them. I think the FIP introducing DataCap removal + using a dealbot or something similar to start tracking retrieval attempts is a good way to start minimizing how much DataCap ends up in the wrong hands. IMO, LDNs likely need to continue moving in the direction of:

- more up front KYC from a client
- less friction to first allocation combined with better tracking of on-chain behavior post allocation
- audit tools that flag risk based on on-chain behavior to prevent future allocation
- and subsequently, removal of DataCap + banning of a client address + associated addresses in the case of abuse

cc @galen-mcandrew @Kevin-FF-USA

**Comment by NSC-FIL** [comment]
I do think Snap deals functionality will be an improvement to Filecoin and appreciate the hard work the devs and others have put into getting this launched.
But the concern that "bad actors" will self-deal and increase the network storage power and unfairly benefit from increased block rewards, while reducing the chance for other SP's is a real possibility.  Is there a way we could have the the Fil+ program estimate the anticipated increase in our network storage power from verified deals before the upgrade happens?  And maybe explain the precautions Fil+ has taken in anticipation of the upgrade and how penalizing "bad actors" would be enforced?  It might help alleviate some concerns.

**Reply by jennijuju** [reply]
Thanks for ack the work and effort being done here!

re your concerns, they seems mostly related to FIL+ governance rather than the upgrade itself (which is what this discussion for). 

This upgrade also introduce a method to remove unused DataCap from bad clients. 

I think you will get better responses and discussions if you repost your questions there https://github.com/filecoin-project/notary-governance/discussions.

**Comment by rjan90** [comment]
First of all, for those voicing their wanted changes / issues about this upgrade. Those comments / thoughts should have been posted during the FIP discussion, and/or draft-process: [https://github.com/filecoin-project/FIPs/discussions](https://github.com/filecoin-project/FIPs/discussions) and not after the FIP has been accepted and on its way to be implemented.

That said, I want to weigh in on the positive side. I think the OhSnap-upgrade is a big improvement for the Filecoin-network as a whole.

- Clients data will be sealed way faster (15-30 minutes, compared to 4-6 hours in a fast sealing pipeline)
- Storage Providers can upgrade CC-sectors more easily then todays time consuming `mark-for-upgrade` process.

About some of the concerns:

- V1-sectors will not be eligible for SnapUpgrades
- Storage Providers has to put up 10x collateral for verified deals, so its not like a Storage Provider can get a lot of power without putting up a lot of collateral to back it.

For the concerns about malicious actors filing for FIL+ datacaps, the OhSnap-upgrade also includes the ability to revoke a verified clients datacap, which should ease some of those concerns. [https://github.com/filecoin-project/FIPs/pull/226](https://github.com/filecoin-project/FIPs/pull/226).
That said, I personally think the governance process can be improved, and recommend that if you have any ideas/thoughts about how to improve, you start a FIP-discussion outlining the idea.

**Comment by Reiers** [comment]
We at **TechHedge** is really excited about this upgrade. 
CC has served its purpose (_at least for us_) and this is the next step. 
_Cant wait to turn the random data into something useful!!_ 

I can see that this discussion derailed a bit from general questions. So I will just add a few answers/highlights to clear things up. 
 - Like a few people have already mention; v1-sectors will **NOT** be eligible for snap-up.
 - You will still need the production line of workers to do `snap-up`. Here are a little peek at resource table from my testing:
 ## 32GiB Sector
 <html>
<body>


Sealing Pipeline | SDR | CUDA | HW | RAM usage
-- | -- | -- | -- | --
AP | ~3 minutes | - | TR Pro 3975wx | -
RU | - | ~5 minutes | FE 3090 | -
PRU2 | - | ~10 minutes | FE 3090 | ~192GB


</body>
</html>

- 10x collateral for FIL+ Deals is still the case.
- DataCap removal will also be introduced in this upgrade.

**Reply by RobQuistNL** [reply]
I agree.

I'm just afraid that it will be abused.
